rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java

```diff
@@ -285,6 +286,10 @@ public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionCon
         }
     }
 
+    private static Predicate<String> classPathMatcher(final String target){
+        return (s) -> s.matches(target.replaceAll("[\\.\\$]", "(\\\\\\.|\\\\\\$)"));
```

Comment:
I was going to suggest:
```suggestion
        return (s) -> s.matches(target.replaceAll("[.$]", "[.\\$]"));
```
but I now wonder if we can't simply do:
```suggestion
            return s -> TypeUtils.fullyQualifiedNamesAreEqual(s, target);
```

Reply:
Interesting, I'll have a look at the second approach